cmd/site-replication.go

```diff
@@ -3723,7 +3723,7 @@ func (c *SiteReplicationSys) SiteReplicationMetaInfo(ctx context.Context, objAPI
 				return info, errSRBackendIssue(err)
 			}
 
-			if meta.lifecycleConfig != nil && meta.lifecycleConfig.HasExpiry() {
+			if meta.lifecycleConfig != nil && meta.lifecycleConfig.HasExpiry() && meta.lifecycleConfig.ExpiryUpdatedAt != nil {
```

Comment:
IIUC, we should copy over ILM expiry rules even when `ExpiryUpdatedAt` is nil. When users upgrade to a MinIO version with this feature (and fix), they will have ILM policies with expiry rules with no `ExpiryUpdatedAt`(i.e `nil`). We should add the `!= nil` check few lines below like so,
```go

if meta.lifecycleConfig.ExpiryUpdatedAt != nil {
    updatedAt = meta.lifecycleConfig.ExpiryUpdatedAt
}
allRules[rule.ID] = madmin.ILMExpiryRule{ILMRule: string(ruleData), Bucket: bucket, UpdatedAt: *(meta.lifecycleConfig.ExpiryUpdatedAt)}
```

Reply:
is that true? @krisis isn't it expected that we send ILM rules only when we have this value correct

Reply:
@shtripat will know best. ILM policies created before the MinIO version which introduced `ExpiryUpdatedAt` field will have this field to be `nil`. Unless this is handled specially during server initialization, we would have to handle this when we 'heal' ILM expiry rules between sites in site-replication.